contracts/test/v0.8/VRFCoordinatorV2.test.ts

```diff
@@ -0,0 +1,269 @@
+    maxGasLimit: number;
+    stalenessSeconds: number;
+    gasAfterPaymentCalculation: number;
+    weiPerUnitLink: BigNumber;
+    minimumSubscriptionBalance: BigNumber;
+  };
+  let c: config;
+
+  beforeEach(async () => {
+    let accounts = await ethers.getSigners();
+    owner = accounts[0];
+    subOwner = accounts[1];
+    subOwnerAddress = await subOwner.getAddress();
+    consumer = accounts[2];
+    random = accounts[3];
+    randomAddress = await random.getAddress();
+    oracle = accounts[4];
+    let ltFactory = await ethers.getContractFactory("LinkToken", accounts[0]);
+    linkToken = await ltFactory.deploy();
+    let bhFactory = await ethers.getContractFactory("BlockhashStore", accounts[0]);
+    blockHashStore = await bhFactory.deploy();
+    let mockAggregatorV3Factory = await ethers.getContractFactory("MockV3Aggregator", accounts[0]);
+    mockLinkEth = await mockAggregatorV3Factory.deploy(0, linkEth);
+    let vrfCoordinatorV2Factory = await ethers.getContractFactory("VRFCoordinatorV2", accounts[0]);
+    vrfCoordinatorV2 = await vrfCoordinatorV2Factory.deploy(
+      linkToken.address,
+      blockHashStore.address,
+      mockLinkEth.address,
+    );
+    await linkToken.transfer(subOwnerAddress, BigNumber.from("1000000000000000000")); // 1 link
+    c = {
+      minimumRequestBlockConfirmations: 1,
+      fulfillmentFlatFeePPM: 0,
+      maxGasLimit: 1000000,
+      stalenessSeconds: 86400,
+      gasAfterPaymentCalculation: 21000 + 5000 + 2100 + 20000 + 2 * 2100 - 15000 + 7315,
+      weiPerUnitLink: BigNumber.from("10000000000000000"),
+      minimumSubscriptionBalance: BigNumber.from("100000000000000000"), // 0.1 link
+    };
+    await vrfCoordinatorV2
+      .connect(owner)
+      .setConfig(
+        c.minimumRequestBlockConfirmations,
+        c.fulfillmentFlatFeePPM,
+        c.maxGasLimit,
+        c.stalenessSeconds,
+        c.gasAfterPaymentCalculation,
+        c.weiPerUnitLink,
+        c.minimumSubscriptionBalance,
+      );
+  });
+
+  it("configuration management", async function () {
+    await expect(
+      vrfCoordinatorV2
+        .connect(subOwner)
+        .setConfig(
+          c.minimumRequestBlockConfirmations,
+          c.fulfillmentFlatFeePPM,
+          c.maxGasLimit,
+          c.stalenessSeconds,
+          c.gasAfterPaymentCalculation,
+          c.weiPerUnitLink,
+          c.minimumSubscriptionBalance,
+        ),
+    ).to.be.revertedWith("Only callable by owner");
+    // Anyone can read the config.
+    const resp = await vrfCoordinatorV2.connect(random).getConfig();
+    assert(resp[0] == c.minimumRequestBlockConfirmations);
+    assert(resp[1] == c.fulfillmentFlatFeePPM);
+    assert(resp[2] == c.maxGasLimit);
+    assert(resp[3] == c.stalenessSeconds);
+    assert(resp[4].toString() == c.gasAfterPaymentCalculation.toString());
+    assert(resp[5].toString() == c.weiPerUnitLink.toString());
+  });
+
+  it("subscription lifecycle", async function () {
```

Comment:
These test cases seem very large. It may be entirely necessary to cover the entire lifecycle, but we should also have happy and sad path unit tests for individual functions in isolation (msg.sender permissions tests, individual reverts, etc).

Reply:
Done